pkg/evaluator/evaluator.go

```diff
@@ -107,7 +107,7 @@ func (e *Evaluator) evalDeclaration(scope *scope, decl *parser.Declaration) Valu
 	if decl.Type() == parser.ANY_TYPE && val.Type() != ANY {
 		val = &Any{Val: val}
```

Comment:
trying to figure out why this nested val does not also need copyOrRef:
```
a := 1
b:any
b = a
a = 2
print a b
```
It prints `2 1` as expected, but from this code, it looks like it would share a reference?

Reply:
I just added:
```go
func TestAnyAssignment(t *testing.T) {
	prog := `
a := 1
b:any
b = a
a = 2
print a b
`
	b := bytes.Buffer{}
	fn := func(s string) { b.WriteString(s) }
	Run(prog, fn)
	want := "1 2\n"
	assert.Equal(t, want, b.String())
}
```
and it fails with:
```
    evaluator_test.go:943: want != got
        "1 2\n"
        "2 2\n"
```

Reply:
same on evy.dev - any idea where you saw it print `2 1` ?

Reply:
and of course I mean `2 1` above...

Reply:
oh, maybe I didn't properly unfix it. I fixed it locally, implemented the test, but thought I unfixed it too.

Checking the test - oops. the test is wrong. I did it slightly differently in the test `print b`, but did a `want := "2\n"` instead of `1`. Ok, so I'm just clumsy, not mad. good to know